rfc-128-continuous-deployment.md

```diff
@@ -0,0 +1,228 @@
+# Continuous Deployment
+
+## Summary
+
+In [RFC 70](https://github.com/alphagov/govuk-rfcs/blob/master/rfc-070-path-towards-continuous-deployment-cd.md#proposal) we committed to fully automatic deployments. The following provides a framework for how we will actually achieve this for all of our existing apps.
+
+Decision: in order to safely enable automatic deployments for each app, we will specify minimum criteria that must first be met. See the criteria described [below](#safety-concerns).
+
+An audit of our apps against these criteria can be found in [Appendix A: Compatibility](rfc-128/compatibility.md).
+
+## Problem
+
+Merging a PR for an application repo [results in a new "release" tag](https://github.com/alphagov/govuk-jenkinslib/blob/30e9392d31b98e5c251ea471d793761748eea1e8/vars/govuk.groovy#L123). We have a [Release](https://github.com/alphagov/release) app to monitor these. Every new release tag [is automatically deployed to our Integration environment](https://github.com/alphagov/govuk-jenkinslib/blob/30e9392d31b98e5c251ea471d793761748eea1e8/vars/govuk.groovy#L134).
+
+This is where the automation ends. In order for a release (or series of releases) to be deployed to the GOV.UK Production environment, a developer with [production access](https://docs.publishing.service.gov.uk/manual/rules-for-getting-production-access.html) must:
+
+- Manually check the release is "OK" in Integration.
+- Manually deploy the release to Staging.
+- Manually check the release is "OK" in Staging.
+- Manually deploy the release to Production.
+
+Having these manual steps has lead to a number of problems.
```

Comment:
I think one we haven't mentioned here which it's worth mentioning is that by enabling CD and having everything deployed automatically, we actually release the value of the work when it's ready; there's no point having lots of PRs merged and sitting in the main branch but not released - which delivers zero value to our users.

Reply:
That's a good point. I don't think I can clearly articulate it as a problem, though, as it's hard to measure how much it's true: are users suffering from the delay? Since a lot of our undeployed PRs are mostly technical changes, it's not clear that they are - we tend to deploy "real" changes with more gusto, because we actually care about them.

I think you make a similar point in https://github.com/alphagov/govuk-rfcs/pull/128#discussion_r510939633 around knowing about bugs sooner. I'd like to focus on that, as it's definitely an omission, and there's a definite loss/gain there.

Is that OK?

Reply:
I think I've addressed this in https://github.com/alphagov/govuk-rfcs/pull/128/commits/cecb8f96e6797d6321563f687881459951b4dea5.